chimerapy/manager.py

```diff
@@ -95,127 +92,85 @@ def __init__(
-    ## Message Reactivity API
+    ## Worker -> Manager Messages
     ####################################################################
 
     async def register_worker(self, request: web.Request):
         msg = await request.json()
+        worker_state = WorkerState.from_dict(msg)
 
-        if msg["register"]:
-            self.workers[msg["id"]] = {
-                "name": msg["name"],
-                "addr": msg["addr"],
-                "http_port": msg["http_port"],
-                "http_ip": msg["http_ip"],
-                "url": f"http://{msg['http_ip']}:{msg['http_port']}",
-                "reported_nodes_server_data": False,
-                "served_nodes_server_data": False,
-                "nodes_status": {},
-                "saving_complete": False,
-                "collection_complete": False,
-                "package_loaded": False,
-            }
-            logger.info(
-                f"Manager registered <Worker id={msg['id']} name={msg['name']}> from {msg['addr']}"
-            )
+        logger.debug(worker_state)
 
-        else:
-            logger.info(
-                f"Manager deregistered <Worker id={msg['id']} name={msg['name']}> from {msg['addr']}"
-            )
-            del self.workers[msg["id"]]
+        self.state.workers[worker_state.id] = worker_state
+        logger.info(
+            f"Manager registered <Worker id={worker_state.id} name={worker_state.name}> from {worker_state.ip}"
+        )
+        logger.debug(f"{self}: WorkerState: {self.state.workers}")
 
         return web.json_response(config.config)
 
-    async def update_nodes_status(self, request: web.Request):
+    async def deregister_worker(self, request: web.Request):
         msg = await request.json()
+        worker_state = WorkerState.from_dict(msg)
 
-        # Updating nodes status
-        self.workers[msg["id"]]["nodes_status"] = msg["nodes_status"]
-        self.workers[msg["id"]]["response"] = True
-
-        logger.debug(f"{self}: Nodes status update to: {self.workers}")
+        logger.info(
+            f"Manager deregistered <Worker id={worker_state.id} name={worker_state.name}> from {worker_state.ip}"
+        )
+        del self.state.workers[worker_state.id]
 
         return web.HTTPOk()
 
-    ####################################################################
-    ## Helper Methods (Front-end)
-    ####################################################################
-
-    def dashboard_dict(self) -> Dict:
-
-        # Convert name to id
-        id_to_name = {
-            n: data["object"].name for n, data in self.graph.G.nodes(data=True)
-        }
-
-        # Add default Manager information
-        network_information = {
-            "ip": self.host,
-            "port": self.port,
-        }
+    async def update_nodes_status(self, request: web.Request):
+        msg = await request.json()
+        worker_state = WorkerState.from_dict(msg)
```

Comment:
Same comments as above. Will this method always succeed? It might be nice to handle errors for something like this to send 400 bad requests instead of 500 internal server errors. Does the framework handle this already?

Reply:
AIOHTTP will handle improperly constructed request, like not containing correct headers and meta data for a valid HHTP request. 

It's up to us to verify that the structure of the data is valid. So yes, I think we should make a new issue and PR to address erroneous messages (as this would require validation in all routes).